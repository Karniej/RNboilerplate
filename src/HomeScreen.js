// @flow

import React from 'react';
import { StyleSheet, Text, View, Button } from 'react-native';

class HomeScreen extends React.Component {
  render() {
    const { navigate } = this.props.navigation;
    return (
      <View style={styles.container}>
        <Text style={styles.welcome}>Welcome to Home Screen</Text>
        <Button title="Goto Second" onPress={() => navigate('Second')} />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
  },
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
  },
});
export default HomeScreen;
