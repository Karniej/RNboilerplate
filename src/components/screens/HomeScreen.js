// @flow

import * as React from 'react';
import { StyleSheet, Text, View, Button } from 'react-native';
import { NavigationActions } from 'react-navigation';

class HomeScreen extends React.Component<{ navigation: any }> {
  static navigationOptions = {
    tabBarLabel: 'Home',
    title: 'Home',
  };

  navigate = () => {
    const navigateToSecondScreen = NavigationActions.navigate({
      routeName: 'Second',
    });
    this.props.navigation.dispatch(navigateToSecondScreen);
  };

  render() {
    return (
      <View style={styles.container}>
        <Text style={styles.welcome}>Welcome to Home Screen</Text>
        <Button title="Goto Second" onPress={() => this.navigate()} />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
  },
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
  },
});
export default HomeScreen;
