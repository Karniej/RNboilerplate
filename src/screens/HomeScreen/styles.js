import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  mainContainer: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
  },
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
  },
  container: {
    flex: 1,
    flexDirection: 'column',
    display: 'none',
  },
  subContainer: {
    flex: 2,
    justifyContent: 'center',
  },
  textStyle: {
    fontSize: 25,
    textAlign: 'center',
    paddingHorizontal: 10,
  },
});
export default styles;
