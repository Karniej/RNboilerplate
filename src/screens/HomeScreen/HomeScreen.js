import * as React from 'react'
import { Text, View } from 'react-native'
import PropTypes from 'prop-types'
import styles from './styles'
import { Header, ContentWrapper, MainButton } from '../../components'
import { colors } from '../../config'

const { container, textStyle, subContainer } = styles
class HomeScreen extends React.Component {
  static navigationOptions = {
    title: 'Home',
    drawerLabel: 'Home',
    headerStyle: {
      backgroundColor: 'black'
    }
  }

  render() {
    const { navigation } = this.props

    return (
      <ContentWrapper>
        <Header
          title='React Native Boilerplate'
        />
        <MainButton
          title='Second Screen'
          routeName='Second'
          navigation={navigation}
        />
        <View
          style={container}
        >
          <View
            style={[subContainer, { backgroundColor: colors.accentColor }]}
          >
            <Text
              style={[textStyle, { color: colors.secondaryColor }]}
            >
              Mobile Discussion Panel
            </Text>
          </View>
          <View
            style={[subContainer, { backgroundColor: colors.secondaryColor }]}
          >
            <Text
              style={[textStyle, { color: colors.primaryColor }]}
            >Are you ready to Rock?
            </Text>
          </View>
        </View>
      </ContentWrapper>
    )
  }
}

HomeScreen.propTypes = {
  navigation: PropTypes.shape({ navigate: PropTypes.func.isRequired }).isRequired
}

export default HomeScreen
