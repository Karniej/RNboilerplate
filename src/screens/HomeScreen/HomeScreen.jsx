import * as React from 'react';
import { Text, View, Button } from 'react-native';
import PropTypes from 'prop-types';
import styles from './styles';

class HomeScreen extends React.Component {
  static navigationOptions = {
    title: 'Home',
    drawerLabel: 'Home',
  };

  render() {
    const { navigate } = this.props.navigation;
    return (
      <View style={styles.container}>
        <Text style={styles.welcome}>Welcome to Home Screen</Text>
        <Button testID="second" title="Goto Second" onPress={() => navigate('Second')} />
      </View>
    );
  }
}

HomeScreen.propTypes = {
  navigation: PropTypes.func.isRequired,
};

export default HomeScreen;
