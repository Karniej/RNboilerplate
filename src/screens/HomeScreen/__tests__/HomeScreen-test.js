import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import HomeScreen from '../HomeScreen';

configure({ adapter: new Adapter() });

test('Renders correctly, expect onpress to be defined, but not called', () => {
  const navigation = { navigate: jest.fn() };
  const onPress = jest.fn();
  const wrapper = shallow(<HomeScreen navigation={navigation} onPress={onPress} />);
  expect(onPress).toBeDefined();
  expect(onPress).not.toBeCalled();
  expect(wrapper).toMatchSnapshot();
});
