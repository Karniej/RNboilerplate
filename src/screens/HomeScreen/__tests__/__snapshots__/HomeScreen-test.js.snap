// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly, expect onpress to be defined, but not called 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HomeScreen
    navigation={
      Object {
        "navigate": [MockFunction],
      }
    }
    onPress={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 20,
              "margin": 10,
              "textAlign": "center",
            }
          }
        >
          Welcome to Home Screen
        </Text>,
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
          testID="second"
          title="Goto Second"
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#00d664",
                "fontSize": 18,
              }
            }
          >
            Goto Second
          </Text>
        </TouchableOpacity>,
      ],
      "style": Object {
        "alignItems": "center",
        "backgroundColor": "#F5FCFF",
        "flex": 1,
        "justifyContent": "center",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": "Welcome to Home Screen",
          "ellipsizeMode": "tail",
          "style": Object {
            "fontSize": 20,
            "margin": 10,
            "textAlign": "center",
          },
        },
        "ref": null,
        "rendered": "Welcome to Home Screen",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "activeOpacity": 0.2,
          "children": <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#00d664",
                "fontSize": 18,
              }
            }
          >
            Goto Second
          </Text>,
          "onPress": [Function],
          "testID": "second",
          "title": "Goto Second",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "Goto Second",
            "ellipsizeMode": "tail",
            "style": Object {
              "color": "#00d664",
              "fontSize": 18,
            },
          },
          "ref": null,
          "rendered": "Goto Second",
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 20,
                "margin": 10,
                "textAlign": "center",
              }
            }
          >
            Welcome to Home Screen
          </Text>,
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
            testID="second"
            title="Goto Second"
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#00d664",
                  "fontSize": 18,
                }
              }
            >
              Goto Second
            </Text>
          </TouchableOpacity>,
        ],
        "style": Object {
          "alignItems": "center",
          "backgroundColor": "#F5FCFF",
          "flex": 1,
          "justifyContent": "center",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "Welcome to Home Screen",
            "ellipsizeMode": "tail",
            "style": Object {
              "fontSize": 20,
              "margin": 10,
              "textAlign": "center",
            },
          },
          "ref": null,
          "rendered": "Welcome to Home Screen",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "activeOpacity": 0.2,
            "children": <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#00d664",
                  "fontSize": 18,
                }
              }
            >
              Goto Second
            </Text>,
            "onPress": [Function],
            "testID": "second",
            "title": "Goto Second",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "Goto Second",
              "ellipsizeMode": "tail",
              "style": Object {
                "color": "#00d664",
                "fontSize": 18,
              },
            },
            "ref": null,
            "rendered": "Goto Second",
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
